api/v1/routes/waitlist.py

```diff
@@ -95,4 +96,19 @@ def admin_add_user_to_waitlist(
         data = {"email": item.email, "full_name": item.full_name},
     )
 
-    return JsonResponseDict(**resp)
\ No newline at end of file
+    return JsonResponseDict(**resp)
+
+@waitlist.get("/users", response_model=success_response, status_code=200)
+async def get_all_waitlist_emails(
+    request: Request,
+    db: Session = Depends(get_db),
+    admin=Depends(get_super_admin)
+):
+    waitlist_users = waitlist_service.fetch_all(db)
+    emails = [user.email for user in waitlist_users]

```

Comment:
return an array of objects containing the email and full_name

Reply:
I did it this way because the github issue i was working from, it just returned an array only containing the emails